libsupport/src/URI.cpp

```diff
@@ -140,12 +143,57 @@ ShouldURLEncode(int c) {
   if (c == '~') {
     return false;
   }
-
-  // We encode whole paths, so in addition to the standard unencoded characters
-  // above, we should not encode '/' either.
+  if (c == ':') {
```

Comment:
nit: can you convert this to a switch statement? At least for the parts that aren't compactly represented as a range?

Reply:
They form several discontinuous segments of ranges. Using switch-case statements to list them out explicitly has better readability. 

```
!
#$
&'()*+,-./
:;
=
?@
[
]
_
~
```

Reply:
I mean simply replace

```
if ('A' <= c && c <= 'Z') {
  return false;
}
....
if (c == '!') {
  return false;
}

if (c == '$') {
  return false;
}
```

with ```
if ('A' <= c && c <= 'Z') {
  return false;
}

switch (c) {
  case '!': 
  case '$':
  ....
    return false;
}
```

> Using switch-case statements to list them out explicitly has better readability.

I'm not sure what you mean here as the current PR doesn't use switch-case statements at all.

Reply:
I have done what you intended. My comment showed up before I push my commit (which is still under testing).